<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Models for amino acid evolution</title>

		<link rel="stylesheet" href="../reveal.js-3.3.0/css/reveal.css">
		<link rel="stylesheet" href="../reveal.js-3.3.0/css/theme/sky.css">
		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../reveal.js-3.3.0/lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../reveal.js-3.3.0/css/print/pdf.css' : '../reveal.js-3.3.0/css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
				<section>
				
				<div style="font-size:80pt">Codon models for protein evolution</div>
				<p><center>Remco Bouckaert</center></p>
				</section>

<section style="font-size:20pt">
<h3>The problem</h3>
<ul>
<li> Given a sequence alignment $D$, what can we say about its evolutionary history
<li> Tree model $T$ represent evolution through time
</ul>
<img width="100%" src="../Auckland2023/alignment0.png" style="border:0px"/>
<ul>
<li class="fragment">Assume sites $D=\{s_1,\ldots,s_k\}$ are independent: $P(D|T)=\prod_{i} P(s_i|T)$
<li class="fragment">Felsenstein's likelihood: 
	<br>$P(s_i|T)=\sum_{x_j}P(x_{j,root}|T)\prod_{j}P(x_j|x_{\pi(x_j)},T)$
	<div class="fragment">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;$=\sum_{x_j}P(x_{j,root})\prod_{j}P(x_j|x_{\pi(x_j)},t_j)$
	<br> where $t_j$ the branch length of node $x_j$
	</div>
</ul>

</section>

<section>
<h2>How sequences evolve</h2>
Instanteneous $20\times20$ rate matrix
<table><tr><td><table>
<tr><td></td><td style="border-left: 2px solid;">$- $</td><td>$ r_{1,2} $</td><td>$ r_{1,3} $</td><td>$ \ldots $</td><td style="border-right: 2px solid;">$r_{1,20}$</td></tr>
<tr><td></td><td style="border-left: 2px solid;">$r_{1,2}  $</td><td>$ - $</td><td>$ r_{2,3} $</td><td>$  \ldots $</td><td style="border-right: 2px solid;">$r_{2,20}$</td></tr>
<tr><td> $R =$ </td><td style="border-left: 2px solid;">$\ldots $</td><td>$ $</td><td>$ $</td><td>$ $</td><td style="border-right: 2px solid;">$ \ldots$</td></tr>
<tr><td></td><td style="border-left: 2px solid;">$\ldots $</td><td>$ $</td><td>$ $</td><td>$ $</td><td style="border-right: 2px solid;">$ \ldots$</td></tr>
<tr><td></td><td style="border-left: 2px solid;">$r_{1,20}  $</td><td>$ r_{2,20} $</td><td>$ $</td><td>$ \ldots $</td><td style="border-right: 2px solid;">$ - $</td></tr>
</table>
</td></tr>
</table>

Symmetric, diagonal = -sum of other elements in row

<div class="fragment">
Frequencies $\pi = \{\pi_1,\ldots,\pi_{20}\}$

<br>$Q = R\times \pi$
</div>
<div class="fragment">
Transition probabilities: $P(t) = e^{-Qt}$
</div>
</section>

<section>
<h2>How sequences evolve</h2>

<center>
<img width="50%" src="../Auckland2023/transitionprobs.png" style="border:0px"/>
<br>time $t$ -->
<br><div class="fragment">$p(x_{j,root}) = \pi_j$</div>
<div style="transform: matrix(0,-1,1,0,-300,-400);">transition probs $p(x_j|x_i,t)$</div>
<div style="transform: matrix(0,-1,1,0,300,-400);">root frequencies $\pi_j$</div>
</center>

<!-->
-->
</section>


<section>
<h2>Empirical models</h2>
Define rate matrix $R$ and frequencies $\pi$
<br >
<ul style="font-size:18pt">
<li>Blosum62
<li>CpREV
<li>DCMut
<li>Dayhoff
<li>FLU
<li>HIVb
<li>HIVw
<li>JTT
<li>LG
<li>MtArt
<li>MtREV
<li>MtMam
<li>RtREV
<li>VT
<li>WAG
<li>$\ldots$
</ul>

</section>


<section>
<h4>Codon models: 61 states</h4>
<center>
		<img width="100%" src="x20.svg" style="border:0px"/>
</center>
<small>Allow 1 nucleotide mutation. &kappa; distinguishes transversions/transitions, &omega; non/synonymous (Goldman &amp; Yang, 1994)</small>
</section>

<section>
<h4>Codon models: Allow all nucleotide mutations.</h4>
<center>
		<img width="100%" src="x202.svg" style="border:0px"/>
</center>

<small> &kappa;<sub style="vertical-align:sub;font-size:12pt">i</sub> distinguishes transversions/transitions, &omega;<sub style="vertical-align:sub;font-size:12pt">i</sub> non/synonymous (based on empirical amino acid model)
(<a href="https://doi.org/10.1093/molbev/msl175">Doron-Faigenboim &amp; Pupko, 2007</a>)
<!--
(<a href="https://doi.org/10.1186/1471-2105-6-134">Schneider et al, 2005</a>)
-->
</small>
</section>

<section>
<h4>Codon models: group amino acids by category.</h4>
<center>
		<img width="100%" src="x203.svg" style="border:0px"/>
</center>

<small> &kappa;<sub style="vertical-align:sub;font-size:12pt">i</sub> distinguishes transversions/transitions, &omega;<sub style="vertical-align:sub;font-size:12pt">i</sub> non/synonymous based on category
</small>
</section>

<section>
<h3>Testing a codon model makes sense</h3>

How: Compare to empirical aminoacid models

$$\psi_i A_{ij}= \sum_{l:aa_l=i} \sum_{s:aa_s=j} \pi_l Q_{ls}$$


<small>
<ul>
<li>
 $\psi_i$ and $\pi_l$ denote the frequency of amino acid $i$ and codon $l$, respectively. </li>
<li>$A_{ij}$ represents the substitution rate from amino acid $i$ to $j$, derived from the empirical amino acid replacement matrix $A$. </li>
<li>$aa_l$ and $aa_s$ denote the amino acids coded by codon $l$ and $s$, respectively, </li>
<li>$Q_{ls}$ entry $l, s$ of codon rate matrix.</li>
</ul>
</small>
Minimise mrsq distance between $\psi_i A_{ij}$ and $\sum_{l:aa_l=i} \sum_{s:aa_s=j} \pi_l Q_{ls}$ by greedy optimisation
</section>



<section>
<h4>Codon model vs empirical AA model: 3 &kappa; only</h4>
<center>
		<img width="100%" src="x3k3f0o.svg" style="border:0px"/>
</center>

</section>


<section>
<h4>Codon model vs empirical AA model: 3 &kappa; + 1 &omega;</h4>
<center>
		<img width="100%" src="x3k3f1o.svg" style="border:0px"/>
</center>
</section>

<section>
<h4>Codon model vs empirical AA model: 3 &kappa; + 5 &omega;</h4>
<center>
		<img width="100%" src="x3k3f5o.svg" style="border:0px"/>
</center>
</section>

<section>
<h4>Codon model vs empirical AA model: 3 &kappa; + 7 &omega;</h4>
<center>
		<img width="100%" src="x3k3f7o.svg" style="border:0px"/>
</center>
</section>

<section>
<h4>Codon model vs empirical AA model: 3 &kappa; + 12 &omega;</h4>
<center>
		<img width="100%" src="x3k3f12o.svg" style="border:0px"/>
</center>
</section>


<section>
<h2>Conclusions</h2>
<ul>
<li>Codon models can capture the essence of empirical AA models
<li>Potential for AARS analysis
<li>...
</ul>
</section>



				</div>
		</div>

		<script src="../Auckland2023/../reveal.js-3.3.0/lib/js/head.min.js"></script>
		<script src="../Auckland2023/../reveal.js-3.3.0/js/reveal.js"></script>
		<!-->
		<script src="../Auckland2023/../reveal.js-3.3.0/plugin/math/math.js"></script>
		-->

		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
					{ src: '../reveal.js-3.3.0/plugin/markdown/marked.js' },
					{ src: '../reveal.js-3.3.0/plugin/markdown/markdown.js' },
					{ src: '../reveal.js-3.3.0/plugin/notes/notes.js', async: true },
					{ src: '../reveal.js-3.3.0/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } }
				],
			});
			Reveal.initialize({
mathjax2: {
    mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@2/MathJax.js',
    config: 'TeX-AMS_HTML-full',
    // pass other options into `MathJax.Hub.Config()`
    tex2jax: {
      inlineMath: [ [ '$', '$' ], [ '\\(', '\\)' ] ],
      skipTags: [ 'script', 'noscript', 'style', 'textarea', 'pre' ]
    }
  },
  plugins: [ RevealMath.MathJax2 ]
			});
		</script>
	</body>
</html>

