<html>
<head>
<style type="text/css">
body{display:inline-block;font:16px arial,sans-serif}
table tr td{padding:1;margin:2;border:2}
a{color:#81b9ed;cursor:pointer}
.talk{color:#41597d;cursor:pointer}
.ttalk{background:#a1b9ed;}
.ttut{background:#f0f0f0;}
</style>
</head>
<body>

<h1>
<img src="../BEAST2Intro/beast.png" width="50"/>
BEAST workshop 3-5th September 2017</h1>
<h2>At the Archipelago Research Institute of University of Turku (Island of Seili, Finland)</h2>

<img src="beastSeili300.png"/>
<img src="Seili.png"/>

<h2>Programme</h2>
<p>This is only a preliminary programme and the titles of talks and contents of tutorial sessions are subject to change. 
<p>
<p>Slides were sourced from many places, including from 
Alexei Drummond,
Carsten Magnus,
Chi Zhang, 
Chieh-Hsi Wu, 
David Rasmussen,
Denise Keuhnert, 
Joelle Barido-Sottani, 
Joseph Heled,
Julija Pecerska, 
Louis du Plessis,
Paul O Lewis,
Quentin Atkinson,
Tim Vaughn, and
Veronika Boskova.
Because it is better to copy the best than to reinvent badly.
<table>
<tr class="tdate"><td colspan="3"><b>Sunday</b> </td></tr>
<tr class="ttalk"><td>17.00</td><td>18.30</td><td><a class="talk" href="1-intro.pdf">Introduction to BEAST</a></td></tr>
<tr class="ttut"><td>18.30</td><td>20.00</td><td>Tutorial <a href="https://taming-the-beast.github.io/tutorials/Introduction-to-BEAST2/">Divergence dating tutorial</a></td></tr>

<tr class="tdate"><td colspan="3"><b>Monday</b></td></tr>
<tr><td>7.00</td><td>8.30</td><td>Breakfast</td></tr>

<tr class="ttalk"><td>8.30</td><td>9.30</td><td>Coalescent/Multi species coalescent</td></tr>
<tr class="ttut"><td>9.30</td><td>11.00</td><td>Tutorial <a href="https://taming-the-beast.github.io/tutorials/species-tree-clocks/species-tree-clocks.pdf">StarBeast2</a></td></tr>

<tr><td>11.00</td><td>12.00</td><td>Lunch </td></tr>

<tr class="ttalk"><td>12.00</td><td>13.00</td><td>Model averaging/selection</td></tr>
<tr class="ttut"><td>13.00</td><td>14.00</td><td>Tutorial <a href="https://taming-the-beast.github.io/tutorials/Substitution-model-averaging/">bModelTest</a></td></tr>

<tr><td>14.00</td><td>14.15</td><td>Coffee break (included)</td></tr>

<tr class="ttalk"><td>14.15</td><td>15.00</td><td>Language models</td></tr>
<tr class="ttut"><td>15.00</td><td>16.00</td><td>Tutorial <a href="https://github.com/rbouckaert/Babel/releases/download/0.0.1/Babel.pdf">Language evolution</a></td></tr>

<tr><td>16.00</td><td>16.45</td><td>Dinner </td></tr>

<tr class="ttalk"><td>16.00</td><td>17.00</td><td>Choosing priors/trouble shooting</td></tr>
<tr class="ttut"><td>17.00</td><td>18.00</td><td>Tutorial <a href="https://github.com/BEAST2-Dev/beast-docs/releases/download/v1.0/BFD-tutorial-2017.zip">Species delimitation</a></td></tr>

<tr><td>19/20</td><td>21/22</td><td>Sauna</td></tr>

<tr><td colspan="3">Light supper served at sauna</td></tr>

<tr class="tdate"><td colspan="3"><b>Tuesday</b></td></tr>

<tr><td>7.00</td><td>8.30</td><td>Breakfast</td></tr>

<tr class="ttalk"><td>8.30</td><td>9.30</td><td>Phylogeography</td></tr>
<tr class="ttut"><td>9.30</td><td>11.00</td><td>Tutorial <a href="https://github.com/BEAST2-Dev/beast-geo/releases/download/v0.0.6/phylogeography_s.v0.1.1.pdf">Spherical diffusion phylogeography</a></td></tr>

<tr><td>11.00</td><td>12.00</td><td>Lunch </td></tr>

<tr class="ttalk"><td>12.00</td><td>12.45</td><td>Cool things to do with BEAST</td></tr>
<tr class="ttut"><td>12.45</td><td>14.00</td><td>BEAST clinic: Bring your own data/problems</td></tr>

<tr><td>14.00</td><td>14.15</td><td>Coffee break </td></tr>

<tr><td>14.15</td><td>15.30</td><td>Wrapping up the course and questions</td></tr>
</table>

<h2>Required Software</h2>

You are required to bring your own laptop to the workshop. The software below must be installed prior to arrival.

<ul>
<li>    Java 8 JDK: This Java SE Developlment Kit is required for BEAST 2 (<a href="http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html">download</a>).
<li>    BEAST 2: BEAST 2 is a cross-platform program for Bayesian phylogenetic analysis of molecular sequences. We will be using this software to perform analyses on the data brought to the workshop. We will be using BEAST v2.4.7 for the workshop (<a href='http://beast2.org/'>download</a>).
<li>    FigTree: FigTree is a graphical viewer of phylogenetic trees, specificly designed to display summarized and annotated trees produced by BEAST (<a href="http://tree.bio.ed.ac.uk/software/figtree/">download</a>).
<li>    Tracer: Tracer is a program for analysing the trace files generated by Bayesian MCMC runs. It can be used to check mixing, ESS, parameter correlations, etc. (<a href="http://tree.bio.ed.ac.uk/software/tracer/">download</a>).
<li>    BEAGLE (optional): BEAGLE is a high-performance library that can perform the core calculations in BEAST2 (and many other phylogenetics packages) much faster. BEAGLE isn't necessary for the workshop, but it will speed up your analyses (<a href="http://beast.bio.ed.ac.uk/beagle">download</a>).
</ul>


<h2>Pre-workshop recommended reading</h2>
Statistics and mathematics primers

We are aware that some of the workshop participants do not have a mathematical background and may have difficulty understanding some of the theory. The two links below can help to refresh the basic concepts, but are by no means meant to be comprehensive resources.

<ul>
<li>    <a href="http://students.brown.edu/seeing-theory/index.html">Seeing Theory</a>: This website gives a very brief, but pretty and easy to understand introduction to many of the fundamental concepts in statistics.
<li>    Ziheng Yang's <a href="http://abacus.gene.ucl.ac.uk/PPS/PrimerProbabilityStatistics.pdf">statistics primer</a>: Gives a more in-depth introduction to the topic.
</ul>

<h2>Phylogenetics and phylodynamics books</h2>

The books below are not all focused on BEAST or even Bayesian phylogenetics. However, each of the books contains at least one background chapter on the theory of molecular evolution, as well as more in-depth chapters on the models implemented in BEAST and BEAST 2. Most of the books also contain at least some discussion of Bayesian inference.

<ul>
<li>    <a href="https://www.beast2.org/book/">Bayesian Evolutionary Analysis with BEAST</a>, Alexei J. Drummond, Remco R. Bouckaert
<li>    <a href="http://abacus.gene.ucl.ac.uk/MESA/">Molecular Evolution: A Statistical Approach</a>, Ziheng Yang
<li>    <a href="http://www.cambridge.org/catalogue/catalogue.asp?isbn=9780521877107">The Phylogenetic Handbook</a>, Philippe Lemey, Marco Salemi, Anne-Mieke Vandamme
</ul>

</body>
</html>